migrations/Version20180810134011.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace Migrations;
+
+use Doctrine\DBAL\DBALException;
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Migrations\IrreversibleMigrationException;
+use Doctrine\DBAL\Schema\Schema;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * Remove nasty hack which adds /java-jars folder on classpath of all java executions.
+ */
+class Version20180810134011 extends AbstractMigration
+{
+  private static $JAVAC_BOX = "javac";
+  private static $JAVA_RUNNER_BOX = "java-runner";
+  private static $JAR_FILES_PORT = "jar-files";
+  private static $FILES_TYPE = "file[]";
+  private static $JARS_BOX_NAME = "jars";
+  private static $PASSED_JARS_BOX_NAME = "jars-passed";
+  private static $FILES_IN_TYPE = "files-in";
+  private static $FILES_OUT_TYPE = "files-out";
+  private static $REMOTE_FILES_TYPE = "remote-file[]";
+
+  /**
+   * @return array java compilation changed pipelines
+   * @throws DBALException
+   */
+  private function updatePipelines(): array {
+    $changedPipelines = [];
+    $pipelinesResult = $this->connection->executeQuery("SELECT * FROM pipeline");
+    foreach ($pipelinesResult as $pipeline) {
+      $pipelineConfig = $this->connection->executeQuery("SELECT * FROM pipeline_config WHERE id = '{$pipeline["pipeline_config_id"]}'")->fetch();
+      $config = Yaml::parse($pipelineConfig["pipeline_config"]);
+      $changed = false;
+
+      foreach ($config["boxes"] as &$box) {
+        if ($box["type"] === self::$JAVAC_BOX) {
+          $changed = true;
+          $changedPipelines[] = $pipeline["id"];
+
+          $box["portsIn"][self::$JAR_FILES_PORT] = [
+            "type" => self::$FILES_TYPE,
+            "value" => self::$JAR_FILES_PORT
+          ];
+          // input jar files in compilation
+          $config["boxes"][] = [
+            "name" => self::$JARS_BOX_NAME,
+            "type" => self::$FILES_IN_TYPE,
+            "portsIn" => [],
+            "portsOut" => [
+              "input" => [
+                "type" => self::$FILES_TYPE,
+                "value" => self::$JAR_FILES_PORT
+              ]
+            ]
+          ];
+          // output jar files from compilation
+          $config["boxes"][] = [
+            "name" => self::$PASSED_JARS_BOX_NAME,
+            "type" => self::$FILES_OUT_TYPE,
+            "portsIn" => [
+              "output" => [
+                "type" => self::$FILES_TYPE,
+                "value" => self::$JAR_FILES_PORT
+              ]
```

Comment:
Sorry, a bit confused on those two boxes here. FILES_IN_TYPE defines portsOut and FILES_OUT_TYPE defines portsIn?

Reply:
yes this is correct